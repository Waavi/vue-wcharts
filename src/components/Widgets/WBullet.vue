<template functional>
    <span
        class="WBullet"
        :style="{
            display: props.display,
            width: `${props.width}px`,
            height: `${props.height}px`,
            background: props.color,
            borderRadius: `${props.borderRadius}%`,
            margin: Array.isArray(props.margin) ? props.margin.map(m => `${m}px`).join(' ') : `${props.margin}px`,
        }"
    />
</template>

<script>
import VueTypes from 'vue-types'

export default {
    name: 'WBullet',
    props: {
        color: VueTypes.string.isRequired,
        display: VueTypes.string.def('inline-block'),
        width: VueTypes.oneOfType([String, Number]).def(10),
        height: VueTypes.oneOfType([String, Number]).def(10),
        borderRadius: VueTypes.oneOfType([String, Number]).def(50),
        margin: VueTypes.oneOfType([VueTypes.number, VueTypes.arrayOf(VueTypes.number).def([0, 0, 0, 0])]).def(8),
    },
}
</script>
