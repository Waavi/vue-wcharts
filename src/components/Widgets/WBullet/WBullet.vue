<template>
    <span
        class="WBullet"
        :style="bulletStylesCmp"
    />
</template>

<script>
import VueTypes from 'vue-types'
import themeMixin from '../../../mixins/theme'

export default {
    name: 'WBullet',
    mixins: [
        themeMixin,
    ],
    props: {
        styles: VueTypes.object.def({}),
    },
    computed: {
        bulletStylesCmp () {
            return {
                ...this.themeStyles.styles,
                ...this.styles,
            }
        },
    },
}
</script>
