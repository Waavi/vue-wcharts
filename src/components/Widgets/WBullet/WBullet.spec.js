import { shallowMount } from '@vue/test-utils'
import WBullet from './WBullet.vue'

describe('Widgets/WBullet', () => {
    const propsData = {
        styles: {
            background: 'black',
        },
    }

    const defaultConfig = {
        propsData,
    }

    xit(`Should be render correctly`, () => {
        const wrapper = shallowMount(WBullet, defaultConfig)
        expect(wrapper.html()).toMatchSnapshot()
    })
})
