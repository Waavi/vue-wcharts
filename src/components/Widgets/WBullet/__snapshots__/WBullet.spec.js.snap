// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/WBullet Should be render correctly 1`] = `<span class="WBullet" style="background: black;"></span>`;
